src/segmentedfilesystem/PacketManager.java

```diff
@@ -0,0 +1,68 @@
+package segmentedfilesystem;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class PacketManager {
+
+    File file1 = new File();
+    File file2 = new File();
+    File file3 = new File();
+
+    public boolean allPackagesReceived(){
+        return file1.fileComplete() && file2.fileComplete() && file3.fileComplete();
+    }
+
+    public void store(PacketStructure packet){
+        if(file1.datapackets.isEmpty()){
+            file1.add(packet);
+        }
+        else if(file2.datapackets.isEmpty()){
+            file2.add(packet);
+        }
+        else if(file3.datapackets.isEmpty()){
+            file3.add(packet);
+        }
+        else{
+            if(packet.getID() == file1.fileID){
+                file1.add(packet);
+            }
+            else if(packet.getID() == file2.fileID){
+                file2.add(packet);
+            }
+            else if(packet.getID() == file3.fileID){
+                file3.add(packet);
+            }
+        }
+    }
+
+    public void fileGenerator() throws IOException {
+        PacketStructure packet;
+
+        ArrayList<Integer> sortedKeys1 = new ArrayList<>(file1.datapackets.keySet());
+        Collections.sort(sortedKeys1);
+        FileOutputStream fos1 = new FileOutputStream(file1.filename);

```

Comment:
You never close any of these output streams, which could cause you to lose some data off the end of the files.